### DISCUSSION TITLE
submit to all three tracks

### DISCUSSION THREAD

**Discussion by rezunli96** [yasserfarouk/scml#60] [General]
Hello. is there a way to submit an agent to all three tracks? Currently what I found is that one can either submit to oneshot, or standard and collusion.

**Comment by yasserfarouk** [comment]
Do you mean the same agent in the three tracks or one agent for oneshot and another for Sydney/collusion?

Technically, it is possible to run the same agent on the three tracks but that uses adapters that are not tested enough so it is not allowed in the official competition this year.

If you are planning to submit two different agents (oneshot and std/collusion), let us know and we will enable it.

**Reply by rezunli96** [reply]
Ahh ok so we do need at least two agents if we want to submit to all three tracks. Currently I do not plan to submit two different agents. But will let you know if needed. Thanks

**Reply by yasserfarouk** [reply]
Ok. Thanks

**Reply by rezunli96** [reply]
Hello, just want to update I would like to submit different agents in different tracks. It would great if such feature is enabled, and perhaps allowing different agent-names for different tracks?

Also just want to confirm: for std/col, it has to be exactly the same agent in these two tracks right? And a participant is automatically enrolled in both tracks if submitted to only one of them?

Thanks!

**Reply by yasserfarouk** [reply]
But in this case, why do not you just register one team per agent? The only thing that is not allowed is to have two agents by the same team **in the same track**.

> Also just want to confirm: for std/col, it has to be exactly the same agent in these two tracks right? And a participant is automatically enrolled in both tracks if submitted to only one of them?

Yes and yes. Nevertheless, only agents with a non-trivial collusion strategy will become finalists in the collusion track.

This is not much of a constraint as the agent can easily find out which track it is running in by counting how many instances of the class are created. In std, it will always be exactly one and in collusion, it will be greater than one. You can refer to [this answer](https://github.com/yasserfarouk/scml/discussions/41#discussioncomment-893531) about how to find out about other agents of the your class in any world simulation.

Until we could find the time and manpower to change the submission system, you can just register up to 3 teams for the 3 tracks. For each of them, just submit the agent for the corresponding track and mark it as incompatible with the other two (the top 3 checkboxes in the `My Agent` page.

**Reply by rezunli96** [reply]
I see. Yes it makes sense to register two teams. Thanks
